lib/saluki-core/src/data_model/event/service_check/mod.rs

```diff
@@ -229,7 +230,10 @@ impl<'a> Serialize for DeduplicatedTagsSerializable<'a> {
     where
         S: Serializer,
     {
-        let deduplicated_tags = self.tags.into_iter().chain(self.origin_tags).deduplicated();
+        let chained_tags = self.tags.into_iter().chain(self.origin_tags);
+
+        let mut tags_deduplicator = ReusableDeduplicator::new();
```

Comment:
We wouldn't get the benefits here right?

Reply:
Unfortunately, yeah: no benefits here.

This is something I hope we can, potentially in the future, replace with a protobuf-based encoder component so that we can match events/metrics.